disperser/cmd/encoder/main.go

```diff
@@ -63,9 +65,31 @@ func RunEncoderServer(ctx *cli.Context) error {
 		logger.Info("Enabled metrics for Encoder", "socket", httpSocket)
 	}
 
+	backendType, err := encoding.ParseBackendType(config.ServerConfig.Backend)
+	if err != nil {
+		return err
+	}
+
 	if config.EncoderVersion == V2 {
+		// Create the encoder
+		opts := []rs.EncoderOption{
+			rs.WithBackend(backendType),
+			rs.WithGPU(config.ServerConfig.GPUEnable),
+		}
+		rsEncoder, err := rs.NewEncoder(opts...)
+		if err != nil {
+			return fmt.Errorf("failed to create encoder: %w", err)
+		}
+
 		// We no longer compute the commitments in the encoder, so we don't need to load the G2 points
-		prover, err := prover.NewProver(&config.EncoderConfig, false)
+		popts := []prover.ProverOption{
+			prover.WithKZGConfig(&config.EncoderConfig),
+			prover.WithLoadG2Points(false),
+			prover.WithBackend(backendType),
+			prover.WithGPU(config.ServerConfig.GPUEnable),
+			prover.WithRSEncoder(rsEncoder),
+		}
+		prover, err := prover.NewProver(popts...)
```

Comment:
Trying to follow options pattern with this change https://golang.cafe/blog/golang-functional-options-pattern.html

Reply:
reverted